modules/data.remote/R/call_MODIS.R

```diff
@@ -111,11 +166,39 @@ call_MODIS <- function(outfolder = ".", start_date, end_date, lat, lon, size = 0
     output$lat <- round(output$lat, 4)
     output$lon <- round(output$lon, 4)
     
-    fname <- paste(product, "_", band, "output_", start_date, "_", end_date, "_", lat, "_", lon, ".csv", sep = "")
-    fname <- paste0(outfolder, "/", fname)
-    write.csv(output, fname)
-    return(output)}
-
+    if (QC_filter == T)
+    {
+      output$qc == as.character(output$qc)
+      for (i in 1:nrow(output))
+      {
+        convert = paste(binaryLogic::as.binary(as.integer(output$qc[i]), n = 8), collapse = "")
+        output$qc[i] = substr(convert, nchar(convert)-2, nchar(convert))
+      }
+      good = which(output$qc == "000" | output$qc == "001")
+      if (length(good) > 0 | !(is.null(good)))
+      {
+        output = output[good,]
+      } else {
+        print("All QC values are bad. No data to output with QC filter == TRUE.")
+      }
+    }
+    
+    if (!(outfolder == "") & !(is.null(siteID)))
+    {
+      fname <- paste(product, "_", band, "_", siteID, "_output_", start_date, "_", end_date, "_", spatial.tools::add_leading_zeroes(iter, 4), ".csv", sep = "")
+      fname <- paste0(outfolder, "/", fname)
+      write.csv(output, fname, row.names = F)
+    }
+    if (!(outfolder == "") & is.null(siteID))
+    {
+      fname <- paste(product, "_", band, "_output_", lat, "_", lon, "_", start_date, "_", end_date, "_", spatial.tools::add_leading_zeroes(iter, 2), ".csv", sep = "")
+      fname <- paste0(outfolder, "/", fname)
```

Comment:
Please use `file.path` here.

Reply:
use file.path where? instead of fname?

Reply:
```suggestion
      fname <- file.path(outfolder, fname)
```

`file.path("a", "b", "c")` will automatically concatenate these with the right separators for the given OS -- i.e. it will create `"a/b/c"` on Unix systems, and `"a\b\c"` on Windows.

Reply:
fixed on my own.... so I'm not committing the suggestion so it doesn't change the other things I fixed recently :). Thanks for the explanation :)